docs/guide/management/sdam-monitoring.txt

```diff
@@ -1,3 +1,438 @@
+      "$clusterTime": {
+        "clusterTime": "6746664774655803393",
+        "signature": {
+          "hash": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=",
+          "keyId": 0
+        }
+      }
+    }
+  }
+
+The type can be one of the following values.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Type
+     - Description
+   * - Unknown
+     - Unknown server
+   * - Standalone
+     - Standalone server
+   * - Mongos
+     - Mongos proxy
+   * - PossiblePrimary
+     - Not checked yet, but another server thinks this is a primary
+   * - RSPrimary
+     - Primary server
+   * - RSSecondary
+     - Secondary server
+   * - RSArbiter
+     - Arbiter
+   * - RSOther
+     - See `the spec <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#rsghost-and-rsother>`_ for more details
+   * - RSGhost
+     - See `the spec <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#rsghost-and-rsother>`_ for more details
+
+serverHeartbeatStarted
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { connectionId: 'localhost:27017' }
+
+serverHeartbeatSucceeded
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { durationMS: 20,
+     reply: {
+       setName: "rs", setVersion: 1, electionId: new ObjectId(),
+       maxBsonObjectSize : 16777216, maxMessageSizeBytes : 48000000,
+       maxWriteBatchSize : 1000, localTime : new Date(),
+       maxWireVersion : 4, minWireVersion : 0, ok : 1,
+       hosts: ["localhost:32000", "localhost:32001"],
+       arbiters: ["localhost:32002"]
+     },
+     connectionId: 'localhost:27017' }
+
+serverHeartbeatFailed
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { durationMS: 20,
+     failure: new MongoError('some error'),
+     connectionId: 'localhost:27017' }
+
+serverOpening
+^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0, name: 'localhost:27017' }
+
+serverClosed
+^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0, name: 'localhost:27017' }
+
+topologyOpening
+^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0 }
+
+topologyClosed
+^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0 }
+
+topologyDescriptionChanged
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   {
+     "topologyId": 0,
+     "previousDescription": {
+       "type": "ReplicaSetWithPrimary",
+       "setName": "rs",
+       "maxSetVersion": 1,
+       "maxElectionId": null,
+       // Note: this field is a Map of server address to serverDescriptions, which
+       // does not serialize in JSON.stringify(). Check out the serverDescriptions
+       // from the serverDescriptionChanged event above to see the contents
+       "servers": {},
+       "stale": false,
+       "compatible": true,
+       "compatibilityError": null,
+       "logicalSessionTimeoutMinutes": 30,
+       "heartbeatFrequencyMS": 30000,
+       "localThresholdMS": 15,
+       "options": {
```

Comment:
we should save `options` to the `TopologyDescription` as a non-enumerable property

Reply:
okay, in the meantime, I'm just listing `options` as an `Object`